web/components/application/hacker/pages/PageTemplate.js

```diff
@@ -46,15 +70,18 @@ export class HackerApplicationPageTemplate extends React.Component {
               onClick={activeIndex === 0 ? cancelApplication : onPageBack}
             />
             <PrimaryButton
-              disabled={!isNextButtonEnabled}
-              text={getPrimaryButtonText(activeIndex, count)}
-              onClick={activeIndex !== count - 1 ? onPageNext : () => this.captcha.execute()}
+              disabled={!isNextButtonEnabled || isSubmitting}
+              text={isSubmitting ? 'Submitting...' : getPrimaryButtonText(activeIndex, count)}
+              onClick={activeIndex !== count - 1 ? onPageNext : () => this.onSubmitButtonClick()}
             />
+            {submitFailed ? (<p>Failed to submit application, please try again!</p>) : null}
             <Reaptcha
               ref={e => (this.captcha = e)}
               sitekey={recaptchaConfig[process.env.NODE_ENV].sitekey}
               onVerify={(token) => {
-                submitApplication(null, token);
+                this.setState({ isSubmitting: true });
```

Comment:
@lynnzhang7 pointed out the delay, and I think that's because we're setting `isSubmitting: true` **after** recaptcha is verified. What do you think about doing this **before** recaptcha is verified, e.g. in https://github.com/nwplus/nwhacks2019/pull/175/files#diff-71f8e35c7373caaf57f69d2b381b566aR29?

Reply:
Ah that's what I initially did, the issue with this is that the case when the puzzle is triggered. If the puzzle was triggered then it would show `Submitting...` while the user is still doing the puzzle (which they can also click out of, leaving the submit button in the `Submitting...` state).